Change GitHub Base URL for octokit

afrocandy opened the pull request:
# Background
I would like to use this action to set a branch protection rules for a repository on GitHub Enterprise server.
But I can't change octokit's GitHub endpoint and also want a mode to set the rules only once.

# Changed
- Be able to change GitHub Base URL with `process.env.GITHUB_API_URL` in each action environment
- Added `add` action (if changing only once)
- Used [request.defaults()](https://github.com/octokit/request.js/#requestdefaults) of `octikit/request.js` to set custom options including token and base url
- Bump up nodejs version 12 to 16 [here](https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/)
- Bump up 1.0.0 to 1.1.0 in package.json

afrocandy commented:
@venh 
Hi, could you check this?

vesubramanian reviewed (commented):
quick question. index.js is auto generated, right? I had only action.js in my original code.

afrocandy commented:
> quick question. index.js is auto generated, right? I had only action.js in my original code.

Yes, index.js was generated by ‘npm run build’.

vesubramanian commented:
But your PR has these files included. Can you please remove those from the PR?

afrocandy commented:
Sorry for late reply and I may have misled you.
Let me explain again.
As I described in description, I also modified some files other than `src/actions.js`.

### 1. dist/index.js
- this file was auto generated by `npm run build`

### 2. action.yml
- modified nodejs version to solve [a warning](https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/)
- added new mode to `action` to want a mode to set the rules only once, so modify description in the file

### 3. README.md
- modified the indent of rules.json sample
- modified the contents because of adding  new mode to `action`

### 4. package.json
- bump up version 1.0.0 to 1.1.0

### 5. package-lock.json
- bump up version 1.0.0 to 1.1.0

Even in this case, should I remove the above 5 files except `src/actions.js` from this PR?
And will you update the above 5 files on your side?

vesubramanian commented:
No @afrocandy. In fact, I am absolutely thankful and grateful for these upgrades and I appreciate. Please remove dist folder and index.js. That should be enough.

afrocandy commented:
> No @afrocandy. In fact, I am absolutely thankful and grateful for these upgrades and I appreciate. Please remove dist folder and index.js. That should be enough.

Thank you for your quick response.
Okay, I'm using this `dist/index.js` on my github action environment now, so I will close this PR and create new PR with removing `dist/index.js` later.

afrocandy commented:
I close this PR because I created new PR with removing `dist/index.js`.

https://github.com/venh/branch-protection/pull/2
